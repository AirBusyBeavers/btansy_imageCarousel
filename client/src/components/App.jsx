import React, { useState, useEffect } from 'react';
import axios from 'axios';
import ImageHeader from './ImageHeader.jsx';

const App = () => {
  const [isShowModal, setIsShowModal] = useState(false);
  const [loadedGallery, setLoadedGallery] = useState({});
  const [dimensions, setDimensions] = useState({
    height: window.innerHeight,
    width: window.innerWidth,
  });

  const resizeHandler = () => {
    setDimensions({
      height: window.innerHeight,
      width: window.innerWidth,
    });
  };

  const viewSelectHandler = (bool) => {
    setIsShowModal(bool);
  };

  useEffect(() => {
    window.addEventListener('resize', resizeHandler);

    return () => window.removeEventListener('resize', resizeHandler);
  });


  const fetchData = (id) => {
    const url = `http://localhost:3000/gallery/${id}`;

    axios.get(url)
      .then(({ data }) => {
        setLoadedGallery(data[0]);
      })
      .catch((error) => {
        console.log(error);
      });
  };


  useEffect(() => {
    fetchData(32);
  }, []);

  console.log(loadedGallery);
  console.log(dimensions);

  let content = <p>Loading...</p>;
  if (!isShowModal && loadedGallery && Object.keys(loadedGallery).length > 0) {
    content = (
      <>
        <ImageHeader
          dimensions={dimensions}
          listingObj={loadedGallery}
        />
      </>
    );
  }
  return content;
};


export default App;
