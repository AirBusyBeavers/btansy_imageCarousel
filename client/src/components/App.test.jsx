/* eslint-env jest */
import React from 'react';
import { shallow, mount } from 'enzyme';
import axios from 'axios';
import App from './App.jsx';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<App />);
  });

  it('renders', () => {
    expect(wrapper).not.toBeNull();
  });
});
