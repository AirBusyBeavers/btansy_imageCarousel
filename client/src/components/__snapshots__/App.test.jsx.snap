// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App should match the snapshot 1`] = `"<p>Loading...</p>"`;

exports[`Unit Tests App should match the snapshot 1`] = `"<p>Loading...</p>"`;
